test/powershell/Modules/Microsoft.PowerShell.Management/FileSystem.Tests.ps1

```diff
@@ -1,230 +1,732 @@
+            ($result.TrimEnd('/\')) | Should Be "$TESTDRIVE"
+        }
+
+        It "Verify Join-Path" {
+            $result = Join-Path -Path "TestDrive:" -ChildPath "temp"
+
+            if ($IsWindows) {
+                $result | Should BeExactly "TestDrive:\temp"
+            }
+            else {
+                $result | Should BeExactly "TestDrive:/temp"
+            }
+        }
+
+        It "Verify Split-Path" {
+            $testPath = Join-Path "TestDrive:" "MyTestFile.txt"
+            $result = Split-Path $testPath -Qualifier
+            $result | Should BeExactly "TestDrive:"
+        }
+
+        It "Verify Test-Path" {
+            $result = Test-Path $HOME
+            $result | Should Be $true
+        }
     }
 }
 
-Describe "Validate basic host navigation functionality" -Tags "CI" {
+Describe "Extended FileSystem Provider Tests" -Tags "Feature" {
     BeforeAll {
+        $testDir = "testDir"
+        $testFile = "testFile.txt"
+        $testFile2 = "testFile2.txt"
         $restoreLocation = Get-Location
-        Set-Location -Path "TestDrive:\"
-
-        #build semi-complex directory structure to test navigation within
-        $level1_0 = "Level1_0"
-        $level2_0 = "Level2_0"
-        $level2_1 = "Level2_1"
-        New-Item -Path $level1_0 -ItemType Directory > $null
-        New-Item -Path (Join-Path $level1_0 $level2_0) -ItemType Directory > $null
-        New-Item -Path (Join-Path $level1_0 $level2_1) -ItemType Directory > $null
     }
 
     AfterAll {
         #restore the previous location
         Set-Location -Path $restoreLocation
     }
-
+    
     BeforeEach {
         Set-Location -Path "TestDrive:\"
     }
 
-    It "Verify Get-Location and Set-Location" {
-        $currentLoc = Get-Location
-        Set-Location $level1_0
-        $level1Loc = Get-Location
-        Set-Location $level2_0
-        $level2Loc = Get-Location
-        $currentLoc.Path | Should Be (Join-Path "TestDrive:" "")
-        $level1Loc.Path | Should Be (Join-Path "TestDrive:" $level1_0)
-        $level2Loc.Path | Should Be (Join-Path (Join-Path "TestDrive:" $level1_0) $level2_0)
-    }
-
-    It "Verify Push-Location and Pop-Location" {
-        #push a bunch of locations
-        Push-Location
-        $push0 = Get-Location
-        Set-Location $level1_0
-        Push-Location
-        $push1 = Get-Location
-        Set-Location $level2_0
-        Push-Location
-        $push2 = Get-Location
-
-        #navigate back home to change path out of all pushed locations
-        Set-Location "TestDrive:\"
-
-        #pop locations off
-        Pop-Location
-        $pop0 = Get-Location
-        Pop-Location
-        $pop1 = Get-Location
-        Pop-Location
-        $pop2 = Get-Location
-
-        $pop0.Path | Should Be $push2.Path
-        $pop1.Path | Should Be $push1.Path
-        $pop2.Path | Should Be $push0.Path
-    }
-}
+    Context "Valdiate New-Item parameters for the FileSystem provider" {
+        AfterEach {
```

Comment:
Fixed.
I went through and factored out almost all of the Before/AfterEach out of the contexts and pushed them to the describe level.